File: tests/test-data/mime-artifact-sample/scripts/hello.pl
#!/usr/bin/env perl
use strict;
use warnings;
use utf8;
use File::Basename qw(dirname);
use File::Spec;
# Hello world using conf/app-config.toml (MIME: text/x-perl)
my $root = $ENV{KAIROS_MIME_SAMPLE_ROOT};
if ( !defined $root || $root eq '' ) {
    my $scripts = dirname( File::Spec->rel2abs($0) );
    $root = dirname($scripts);
}
open my $fh, '<:encoding(UTF-8)', "$root/conf/app-config.toml" or die "open app-config.toml: $!\n";
my $msg;
while ( my $line = <$fh> ) {
    if ( $line =~ /^\s*message\s*=\s*"([^"]*)"/ ) {
        $msg = $1;
        last;
    }
}
close $fh;
print( ( $msg // 'missing message' ), "\n" );
exit( defined $msg && $msg ne '' ? 0 : 1 );

File: tests/test-data/mime-artifact-sample/scripts/hello.py
#!/usr/bin/env python3
"""Hello world using conf/app-config.toml (MIME: text/x-python)."""
from __future__ import annotations

import os
import re
import sys
from pathlib import Path

_SAMPLE_ROOT = Path(__file__).resolve().parent.parent
ROOT = Path(os.environ.get("KAIROS_MIME_SAMPLE_ROOT", _SAMPLE_ROOT))


def read_toml_message(text: str) -> str:
    m = re.search(r'^\s*message\s*=\s*"([^"]*)"', text, re.MULTILINE)
    return m.group(1) if m else ""


def main() -> int:
    path = ROOT / "conf" / "app-config.toml"
    msg = read_toml_message(path.read_text(encoding="utf-8"))
    print(msg or "missing message in conf/app-config.toml", file=sys.stdout)
    return 0 if msg else 1


if __name__ == "__main__":
    raise SystemExit(main())